packages/lb-annotation/src/core/pointCloud/index.ts

```diff
@@ -1144,6 +1213,49 @@ export class PointCloud {
     };
   }
 
+  public getSpherePoint2DCoordinate(sphereParams: IPointCloudSphere) {
+    const { center, radius } = sphereParams;
+    const transParams = { center, width: radius * 2, height: radius * 2, depth: radius * 2, rotation: 0 };
+    const projectMatrix = new THREE.Matrix4()
+      .premultiply(this.camera.matrixWorldInverse) // View / Camera Translation
+      .premultiply(this.camera.projectionMatrix); // Projection Translation
+
+    const boxSideMatrix = new THREE.Matrix4()
+      .premultiply(this.getModelTransformationMatrix(transParams)) // Model Translation
+      .premultiply(projectMatrix) // View Translation + Projection Translation
+      .premultiply(this.basicCoordinate2CanvasMatrix4); // Viewport Translation
+    this.sideMatrix = boxSideMatrix;
+
+    const point2d = new THREE.Vector3(center.x, center.y, center.z).applyMatrix4(this.sideMatrix as any);
+
+    const wZoom = this.containerWidth / (radius * 2);
+    const hZoom = this.containerHeight / (radius * 2);
+
+    return {
+      point2d,
+      zoom: Math.min(wZoom, hZoom) / 2,
+    };
+  }
+
+  public getSphereTopPoint2DCoordinate(sphereParams: IPointCloudSphere) {
+    const { radius, center } = sphereParams;
+    const transParams = { center, width: radius * 2, height: radius * 2, depth: radius * 2, rotation: 0 };
+
+    const vector = new THREE.Vector3(center?.x, center?.y, center?.z).applyMatrix4(
```

Comment:
Don't need to use it, you can delete it. Point is just need to transform `viewport`

Reply:
done